components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -154,6 +172,56 @@ open class GleanInternalAPI internal constructor () {
         return uploadEnabled
     }
 
+    /**
+     * Handles the changing of state when uploadEnabled changes.
+     *
+     * When disabling, all pending metrics, events and queued pings are cleared.
+     *
+     * When enabling, the core Glean metrics are recreated.
+     *
+     * If Glean is not initialized, this is a no-op.
```

Comment:
That is not true. 
Neither `initializeCoreMetrics` nor `clearMetrics` have this check.
It's only call site (for now) has the check.

Reply:
Yeah, I think we just remove this comment, then.  It's a private method, and all callers of this method ensure that invariant.